tests/unit/plugins/kubernetes/kubernetes_pod_executor_test.py

```diff
@@ -266,3 +281,75 @@ def test_process_event_enqueues_task_processing_events_deleted(
 
     assert k8s_executor.event_queue.qsize() == 1
     assert len(k8s_executor.task_metadata) == 0
+
+
+def test_reconcile_missing_pod(
+    k8s_executor,
+):
+    with mock.patch.object(k8s_executor, "kube_client", autospec=True) as mock_kube_client:
+        mock_kube_client.get_pod.return_value = None
+    task_config = mock.Mock(spec=KubernetesTaskConfig)
+    task_config.name = 'job_1.action_1'
+    task_config.pod_name = 'pod_name'
+
+    k8s_executor.task_metadata = pmap({
+        task_config.pod_name: KubernetesTaskMetadata(
+            task_config=mock.Mock(spec=KubernetesTaskConfig),
+            task_state=KubernetesTaskState.TASK_UNKNOWN,
+            task_state_history=v(),
+        )
+    })
+
+    k8s_executor.reconcile(task_config)
+
+    assert k8s_executor.event_queue.qsize() == 1
+    assert len(k8s_executor.task_metadata) == 1
+    tm = k8s_executor.task_metadata['pod_name']
+    assert tm.task_state == KubernetesTaskState.TASK_LOST
+
+
+def test_initial_task_metadata(
+    k8s_executor_with_tasks
+):
+    assert 'pod_1' in k8s_executor_with_tasks.task_metadata
+    assert all(
+        [
+            tm.task_state == KubernetesTaskState.TASK_UNKNOWN
+            for tm in k8s_executor_with_tasks.task_metadata.values()
+        ]
+    )
+
+
+def test_reconcile_running_pod(
+    k8s_executor_with_tasks
```

Comment:
thoughts on having the fixture yield a tuple of (executor, task_configs)?

Reply:
This worked nicely, and it let me test all 4 pod phases we handle much easier.